File: src/main/java/com/sotasan/decompiler/menus/file/File.java
package com.sotasan.decompiler.menus.file;

import com.formdev.flatlaf.extras.components.FlatMenu;
import com.sotasan.decompiler.services.LanguageService;
import javax.swing.*;
import java.awt.event.KeyEvent;

public class File extends FlatMenu {

    public File() {
        setMnemonic(KeyEvent.VK_F);
        setText(LanguageService.getTranslation("file"));

        add(new FileOpenFile());
        add(FileCloseTab.getINSTANCE());
        add(new JSeparator());
        add(new FileNewInstance());
        add(new FileExit());
    }

}

File: src/main/kotlin/moe/sota/decompiler/menus/file/File.kt
package moe.sota.decompiler.menus.file

import com.formdev.flatlaf.extras.components.FlatMenu
import moe.sota.decompiler.services.LanguageService
import java.awt.event.KeyEvent
import javax.swing.JSeparator

class File(
    openFile: FileOpenFile,
    findFile: FileFind,
    searchFile: FileSearch,
    closeTab: FileCloseTab,
    newInstance: FileNewInstance,
    exit: FileExit
) : FlatMenu() {

    init {
        mnemonic = KeyEvent.VK_F
        text = LanguageService.getTranslation("file")

        add(openFile)
        add(findFile)
        add(searchFile)
        add(closeTab)
        add(JSeparator())
        add(newInstance)
        add(exit)
    }

}
